pallets/nomad/da-bridge/src/lib.rs

```diff
@@ -102,11 +108,9 @@ pub mod pallet {
 		fn do_dispatch_data_root(
 			destination_domain: u32,
 			recipient_address: H256,
-			header: T::Header,
+			block_number: T::BlockNumber,
+			data_root: H256,
```

Comment:
data_root is completely unlinked to header hash and block number. This means a bad actor could just make up some arbitrary data root as long as it found a header hash and block number that work

Reply:
I cannot find a simple way to access to the header info from inside the runtime, so I've reverted this change. The whole header must be send.